OlinkAnalyze/R/check_npx_col_names.R

```diff
@@ -0,0 +1,242 @@
+#' of exporting data with different quantification approaches. This function
+#' aims to instruct each function of OlinkAnalyze on the column it should be
+#' using for the downstream analysis. This should be seamless for data exported
+#' from Olink Software and imported to R using the read_npx function.
+#'
+#' However, in certain cases the columns of interest might be named differently.
+#' This function allows assigning custom-named columns of a data frame to
+#' internally expected variables that will in turn instruct Olink Analyze
+#' functions to use them for downstream analysis. For example, if one has
+#' transformed the existing NPX value and has stored the result in a new column
+#' called NPX_2, then they can assign this new name to the internal variable
+#' `quant` to inform the package that in the downstream analysis `NPX_2` should
+#' be used. See example 1.
+#'
+#' Similarly, in case of multiple matches (e.g. the data frame contains both
+#' columns `NPX` and `Quantified_value`) the ties will need to be resolved by
+#' the user using the argument `preferred_names` from this function.  See
+#' example 2.
+#'
+#' The argument `preferred_names` is a named character vector with internal
+#' column names as names and column names of the current data set as values.
+#' Names of the input vector can be one or more of the following: `sample_id`,
+#' `sample_type`, `olink_id`, `plate_id`, `qc_warning`, `lod`, `quant`.
+#'
+#' @author
+#'  Klev Diamanti
+#'  Masoumeh Sheikhi
+#'
+#' @param df A tibble or and arrow object from `read_npx`.
+#' @param preferred_names A named character vector where names are internal
+#' column names and values are column names to be selected from current data
+#' frame. Read the `description` for further information.
+#'
+#' @return A list of matched column names based on the preferred names.
+#'
+#' @examples
+#' \donttest{
+#' file <- system.file("extdata",
+#'                     "npx_data_ext.parquet",
+#'                      package = "OlinkAnalyze")
+#' df <- read_npx(filename = file)
+#'
+#' # run df as is
+#' OlinkAnalyze:::check_npx_col_names(df = df)
+#'
+#' # SampleType missing
+#' df |>
+#'   dplyr::select(
+#'     -dplyr::all_of("SampleType")
+#'   ) |>
+#'   OlinkAnalyze:::check_npx_col_names()
+#'
+#' # use PCNormalizedNPX instead on NPX
+#' OlinkAnalyze:::check_npx_col_names(
+#'   df = df,
+#'   preferred_names = c("quant" = "PCNormalizedNPX")
+#' )
+#'
+#' # use PCNormalizedNPX instead on NPX, and PlateLOD instead of LOD
+#' df |>
+#'   dplyr::mutate(
+#'     LOD = 1L,
+#'     PlateLOD = 2L
+#'   ) |>
+#'   OlinkAnalyze:::check_npx_col_names(
+#'     preferred_names = c("quant" = "PCNormalizedNPX",
+#'                         "lod" = "PlateLOD")
+#'   )
+#' }
+#'
+check_npx_col_names <- function(df,
+                                preferred_names = NULL) {
+
+  # check input ----
+
+  check_is_arrow_or_tibble(df = df,
```

Comment:
Shall we remove this line and change some of the unit tests to tibble instead of arrow_table?